File: Solutions/mario/mario.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
    if (argc != 2)
	{
		printf("Usage: ./mario integer between 1 and 8\n");
		return 1;
	}

    int number = atoi(argv[1]);

	if (number < 1 || number > 8)
    {
        printf("Usage: ./mario	integer between 1 and 8\n");
		return 1;
    }

    for (int i = 0; i < number; i++)
    {

		for (int j = number-i; j > 1; j--)
		{
			printf(" ");
		}

		for (int j = 0; j < i+1; j++)
		{
			printf("#");
		}

		printf("  ");

		for (int j = 0; j < i+1; j++)
		{
			printf("*");
		}

		printf("\n");

	}

	return 0;

}

File: Solutions/mario/mario.py
def main():
    
    height = get_int("Height: ")
    
    while height < 1 or height > 8:
        print("Choose a number between 1 and 8")
        height = get_int("Height: ")

    for i in range(height):
        
        empty_line = ''
        bricks = ''

        counter = 0
        while counter < height - i - 1:
            empty_line += ' '
            counter += 1

        for j in range(i+1):
            bricks += '#'

        print(empty_line + bricks + '  ' + bricks)


def get_int(prompt):
    while True:
        try:
            return int(input(prompt))
        except ValueError:
            print("Not an integer")
            

main()
